src/mjolnir/CMakeLists.txt

```diff
@@ -25,12 +25,34 @@ if (UNIX)
   # Enables stricter compiler checks on a file-by-file basis
   # which allows us to migrate piecemeal
   set_source_files_properties(
+    bssbuilder.cc
+    complexrestrictionbuilder.cc
+    countryaccess.cc
+    dataquality.cc
+    directededgebuilder.cc
+    graphtilebuilder.cc
+    edgeinfobuilder.cc
+    elevationbuilder.cc
     graphenhancer.cc
+    graphvalidator.cc
     hierarchybuilder.cc
+    linkclassification.cc
     luatagtransform.cc
+    node_expander.cc
     osmdata.cc
+    osmpbfparser.cc
+    osmaccessrestriction.cc
     osmrestriction.cc
-    PROPERTIES COMPILE_FLAGS "-Werror -Wall")
+    osmway.cc
+    pbfadminparser.cc
+    restrictionbuilder.cc
+    servicedays.cc
+    shortcutbuilder.cc
+    timeparsing.cc
+    transitbuilder.cc
+    util.cc
+    validatetransit.cc
+    PROPERTIES COMPILE_FLAGS "-Werror -Wall -Werror=sign-compare")
```

Comment:
I believe clang doesn't include `sign-compare` warning to the `-Wall` I had to add it manually to reproduce errors in CI.
I see 2 options here:
- `-Werror=sign-compare`(in the PR at the moment)
- `-Wno-error=sign-compare`, i.e don't treat this warning as error

Reply:
CI isnt building with clang except for on osx though (unless that changed...?)